src/core/ext/transport/chttp2/transport/parsing.cc

```diff
@@ -943,11 +943,9 @@ grpc_error_handle grpc_chttp2_header_parser_parse(void* hpack_parser,
   if (s != nullptr) {
     s->stats.incoming.header_bytes += GRPC_SLICE_LENGTH(slice);
 
-    if (s->context != nullptr) {
-      call_tracer = static_cast<grpc_core::CallTracerAnnotationInterface*>(
-          static_cast<grpc_call_context_element*>(
-              s->context)[GRPC_CONTEXT_CALL_TRACER_ANNOTATION_INTERFACE]
-              .value);
+    if (s->arena != nullptr) {
```

Comment:
In what case will `s->arena` be null?  Right offhand, it seems like that shouldn't happen, but maybe I'm missing something.